unstructured/ingest/connector/github.py

```diff
@@ -0,0 +1,188 @@
+        return Path(self.config.output_dir) / f"{self.path}.json"
+
+    def _create_full_tmp_dir_path(self):
+        """includes "directories" in s3 object path"""
+        self.filename.parent.mkdir(parents=True, exist_ok=True)
+
+    def cleanup_file(self):
+        """Removes the local copy the file (or anything else) after successful processing.
+        Not relevant for GitHubIngestDoc."""
+        if not self.config.preserve_downloads:
+            if self.config.verbose:
+                print(f"cleaning up {self}")
+            os.unlink(self.filename)
+
+    def get_file(self):
+        """Fetches the "remote" doc and stores it locally on the filesystem."""
+        self._create_full_tmp_dir_path()
+        if not self.config.re_download and self.filename.is_file() and self.filename.stat():
+            if self.config.verbose:
+                print(f"File exists: {self.filename}, skipping download")
+            return
+
+        if self.config.verbose:
+            print(f"fetching {self} - PID: {os.getpid()}")
+        content_file = self.repo.get_contents(self.path)
+        with open(self.filename, "wb") as f:
+            f.write(content_file.decoded_content)  # type: ignore
+
+    def has_output(self):
+        """Determine if structured output for this doc already exists."""
+        output_filename = self._output_filename()
+        return output_filename.is_file() and output_filename.stat()
+
+    def write_result(self):
+        """Write the structured json result for this doc. result must be json serializable."""
+        output_filename = self._output_filename()
+        output_filename.parent.mkdir(parents=True, exist_ok=True)
+        with open(output_filename, "w", encoding="utf8") as output_f:
+            json.dump(self.isd_elems_no_filename, output_f, ensure_ascii=False, indent=2)
+        print(f"Wrote {output_filename}")
+
+
+class GitHubConnector(BaseConnector):
+    def __init__(self, config: SimpleGitHubConfig):
+        from github import Github
+
+        self.config = config
+        self.github = Github(self.config.github_access_token)
+        self.cleanup_files = not config.preserve_downloads
+
+    def cleanup(self, cur_dir=None):
+        if not self.cleanup_files:
+            return
+
+        if cur_dir is None:
+            cur_dir = self.config.download_dir
+        sub_dirs = os.listdir(cur_dir)
+        os.chdir(cur_dir)
+        for sub_dir in sub_dirs:
+            # don't traverse symlinks, not that there every should be any
+            if os.path.isdir(sub_dir) and not os.path.islink(sub_dir):
+                self.cleanup(sub_dir)
+        os.chdir("..")
+        if len(os.listdir(cur_dir)) == 0:
+            os.rmdir(cur_dir)
+
+    def initialize(self):
+        pass
+
+    def is_file_type_supported(self, path: str) -> bool:
+        # Workaround to ensure that auto.partition isn't fed with .yaml, .py, etc. files
+        # TODO: What to do with no filenames? e.g. LICENSE, Makefile, etc.
```

Comment:
I assume you mean "no filename extensions." Actually, libmagic does a reasonably good job of guessing them:

```
(unst) bash-5.1$ cp example-docs/fake-html.html example-docs/fake-h
(unst) bash-5.1$ cp example-docs/fake-power-point.pptx example-docs/fake-p
(unst) bash-5.1$ PYTHONPATH=. python -c "import sys; from unstructured.file_utils.filetype import detect_filetype; print(detect_filetype(sys.argv[1]))" example-docs/fake-h
FileType.HTML
(unst) bash-5.1$ PYTHONPATH=. python -c "import sys; from unstructured.file_utils.filetype import detect_filetype; print(detect_filetype(sys.argv[1]))" example-docs/fake-p
FileType.PPTX
```

which means `unstructured.partition.auto::partition()` would do the right thing with them.

Perhaps the check should be for [FileType.UNK](https://github.com/Unstructured-IO/unstructured/blob/9062d25/unstructured/file_utils/filetype.py#L221) ?

Reply:
Whoops, I did mean for no extension. I could rely on libmagic here, but this filtering is applied solely on the file paths: the actual documents have yet to be downloaded: that is done in the `GitHubIngestDoc.get_file` method. Through this filtering, less `GitHubIngestDoc` instances need to be created, and thus less API requests to GitHub are made.

What do you suggest? I can only check for `FileType.UNK` in `GitHubIngestDoc`.